lrlex/src/lib/lexer.rs

```diff
@@ -281,13 +284,37 @@ impl<
                         }
                     }
                 }
-                if let Some(target_state_id) = r.target_state {
-                    state = match self.get_start_state_by_id(target_state_id) {
+                if let Some((target_state_id, op)) = &r.target_state {
+                    // TODO Handle state stack operations
+                    let state = match self.get_start_state_by_id(*target_state_id) {
                         None => {
                             lexemes.push(Err(LexError::new(Span::new(old_i, old_i))));
                             break;
                         }
                         Some(state) => state,
+                    };
+                    let head = state_stack.last_mut();
+                    match op {
+                        StartStateOperation::ReplaceStack => {
+                            state_stack.clear();
+                            state_stack.push((1, state));
+                        },
+                        StartStateOperation::Push => {
+                            match head {
+                                Some((count, s)) if s.id == state.id => *count += 1,
+                                _ => state_stack.push((1, state)),
+                            }
+                        },
+                        StartStateOperation::Pop => {
+                            match head {
+                                Some((count, s)) if *count > 1 => {*count -= 1;},
+                                Some(_) => {state_stack.pop();},
```

Comment:
Should this be something to the effect of:

```
state_stack.pop();
if state_stack.is_empty() { state_stack.push((1, initial_state)) }
```

So there is always an entry in the state_stack?

Reply:
So, I guess that is a real question of whether we should fall back to the initial state on an empty stack, or throw an error when there is an empty stack.  I feel, like with the default behavior being replace, it is going to be very easy to `ReplaceStack` and then `Pop`,

Here https://github.com/ratmice/grmtools/commit/0bc7c1c682e26315af40a34bfb976be2d38a7b72 was a second attempt to simplify things around this, by using `.unwrap_or(initial_state)`: 

The nice thing about it I guess is if you aren't using start states, you won't need to allocate the stack,
I'm not sure if this leads to any win in regards to not having to go to the heap if start_stack is empty.

Anyhow I think you should take this suggestion as something that we may need to decide between producing an error on an empty stack is the right behavior or not.

Reply:
I think as per https://github.com/softdevteam/grmtools/pull/341#issuecomment-1277830747
we can disregard my suggestions here, and this can just produce a `LexError { lexing_state: None, ...}`